src/awala/routes/awala.routes.spec.ts

```diff
@@ -71,49 +91,81 @@ describe('awala routes', () => {
     expect(response).toHaveProperty('statusCode', HTTP_STATUS_CODES.UNSUPPORTED_MEDIA_TYPE);
   });
 
+  test('Missing headers should resolve into bad request', async () => {
+    const response = await server.inject({
+      method: 'POST',
+      url: '/',
+    });
+
+    expect(response.statusCode).toBe(HTTP_STATUS_CODES.BAD_REQUEST);
+  });
+
+  test('Invalid message should should resolve into bad request', async () => {
+    const cloudEvent = new CloudEvent({
+      id: CE_ID,
+      source: AWALA_PEER_ID,
+      type: 'invalid message type',
+      subject: 'https://relaycorp.tech/awala-endpoint-internet',
+      datacontenttype: 'application/vnd.veraid.member-bundle-request',
+    });
+
+    const response = await postEvent(cloudEvent, server);
+
+    expect(response).toHaveProperty('statusCode', HTTP_STATUS_CODES.BAD_REQUEST);
+    expect(logs).toContainEqual(
+      partialPinoLog('error', 'Refused invalid type', {
+        parcelId: CE_ID,
+      }),
+    );
+  });
+
   describe('Member bundle request', () => {
-    const validPayload = {
+    const expiry = addDays(Date.now(), 5);
+    const validMessageContent = {
       publicKeyId: MEMBER_PUBLIC_KEY_MONGO_ID,
       memberBundleStartDate: '2023-04-13T20:05:38.285Z',
-      peerId: AWALA_PEER_ID,
       signature: SIGNATURE,
     };
-    const validHeaders = {
-      // eslint-disable-next-line @typescript-eslint/naming-convention
-      'content-type': 'application/vnd.veraid.member-bundle-request',
-    };
+
+    const cloudEvent = new CloudEvent({
+      id: CE_ID,
+      source: AWALA_PEER_ID,
+      type: INCOMING_SERVICE_MESSAGE_TYPE,
+      subject: 'https://relaycorp.tech/awala-endpoint-internet',
+      datacontenttype: 'application/vnd.veraid.member-bundle-request',
+      expiry: formatISO(expiry),
+      data: JSON.stringify(validMessageContent),
+    });
 
     test('Valid data should be accepted', async () => {
-      const response = await server.inject({
-        method: 'POST',
-        url: '/awala',
-        headers: validHeaders,
-        payload: validPayload,
-      });
       mockCreateMemberBundleRequest.mockResolvedValueOnce({
         didSucceed: true,
       });
 
+      const response = await postEvent(cloudEvent, server);
+
       expect(response).toHaveProperty('statusCode', HTTP_STATUS_CODES.ACCEPTED);
-      expect(mockCreateMemberBundleRequest).toHaveBeenCalledOnceWith(validPayload, {
-        logger: server.log,
-        dbConnection: server.mongoose,
-      });
+      expect(mockCreateMemberBundleRequest).toHaveBeenCalledOnceWith(
+        { ...validMessageContent, peerId: AWALA_PEER_ID },
+        {
+          logger: expect.anything(),
```

Comment:
The problem with this is that we are passing a child instance of the logger to the method. Not sure how to validate that

Reply:
Indeed. It's not impossible... but the solution is really cumbersome. So let's leave this as it is.